src/localizers/moment.js

```diff
@@ -8,12 +8,16 @@ function inSame12Hr(start, end){
   return (s <= 0 && e <= 0) || (s >= 0 && e >= 0)
 }
 
+function shouldTrimAMPM(start, local, culture) {
+  return /(AM|PM)$/i.test(local.format(start, 'LT', culture));
+}
+
 let dateRangeFormat = ({ start, end }, culture, local)=>
   local.format(start, 'L', culture) + ' — ' + local.format(end, 'L', culture)
 
 let timeRangeFormat = ({ start, end }, culture, local)=>
-  local.format(start, 'h:mma', culture) +
-    ' — ' + local.format(end, inSame12Hr(start, end) ? 'h:mm' : 'h:mma', culture)
+  local.format(start, 'LT', culture) +
+    ' — ' + local.format(end, (inSame12Hr(start, end) && shouldTrimAMPM(start, local, culture)) ? 'h:mm' : 'LT', culture)
```

Comment:
> The previous format would force everyone to submit to the AM/PM format which is really terrible localization. I changed very little compared to the original state. The only difference being that we that don't use the AM/PM system won't have it "forced".

they will have some other things forced tho by the h:mm format, That's all to say that you are correct, this whole choice to try and trim the AM/PM bits is a bad choice for a default. Which is why i'm saying lets go farther than this PR, and remove the trimming logic completely, and just use generic "time" format regardless of where it falls, in this case `"LT"`